src/main/java/com/celements/ressource_url/RessourceUrlService.java

```diff
@@ -0,0 +1,185 @@
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Requirement;
+import org.xwiki.model.reference.DocumentReference;
+
+import com.celements.filebase.IAttachmentServiceRole;
+import com.celements.model.access.IModelAccessFacade;
+import com.celements.model.access.exception.AttachmentNotExistsException;
+import com.celements.model.access.exception.DocumentNotExistsException;
+import com.celements.model.context.ModelContext;
+import com.celements.model.util.ModelUtils;
+import com.celements.web.service.LastStartupTimeStampRole;
+import com.google.common.base.Suppliers;
+import com.xpn.xwiki.doc.XWikiAttachment;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.web.XWikiURLFactory;
+
+public class RessourceUrlService implements RessourceUrlServiceRole {
+
+  private static final String ATTACHMENT_LINK_REGEX = "([\\w\\-]*:)?([\\w\\-]*\\.[\\w\\-]*){1};.*";
+  private static final Supplier<Pattern> ATTACHMENT_LINK_PATTERN = Suppliers
+      .memoize(() -> Pattern.compile(ATTACHMENT_LINK_REGEX));
+  private static final String ON_DISK_LINK_REGEX = "^:[/\\w\\-\\.]*";
+  private static final Supplier<Pattern> ON_DISK_LINK_PATTERN = Suppliers
+      .memoize(() -> Pattern.compile(ON_DISK_LINK_REGEX));
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RessourceUrlService.class);
+
+  @Requirement
+  private ModelContext context;
+
+  @Requirement
+  private IModelAccessFacade modelAccess;
+
+  @Requirement
+  private ModelUtils modelUtils;
+
+  @Requirement
+  private IAttachmentServiceRole attachmentSrv;
+
+  @Requirement
+  private LastStartupTimeStampRole lastStartupTimeStamp;
+
+  @Override
+  @NotNull
+  public String getAttachmentName(@NotNull String link) {
+    return link.split(";")[1];
+  }
+
+  @Override
+  @NotNull
+  public DocumentReference getPageDocRef(@NotNull String link) {
+    return modelUtils.resolveRef(link.split(";")[0], DocumentReference.class);
+  }
+
+  @Override
+  public boolean isAttachmentLink(String link) {
+    if (link != null) {
+      return ATTACHMENT_LINK_PATTERN.get().matcher(link.trim()).matches();
+    }
+    return false;
+  }
+
+  @Override
+  public boolean isOnDiskLink(@Nullable String link) {
+    if (link != null) {
+      return ON_DISK_LINK_PATTERN.get().matcher(link.trim()).matches();
+    }
+    return false;
+  }
+
+  private String getAction(Optional<String> action) {
+    if (action.isPresent()) {
+      return action.get();
+    }
+    return getDefaultAction();
+  }
+
+  private String getDefaultAction() {
+    return context.getXWikiContext().getWiki().getXWikiPreference("celdefaultAttAction",
+        "celements.attachmenturl.defaultaction", "file", context.getXWikiContext());
+  }
+
+  @Override
+  @NotEmpty
+  public String getRessourceURLPrefix() {

```

Comment:
naming:
```suggestion
  public String getRessourceUrlPrefix() {
```
change everywhere accordingly